dozer-types/src/types/field.rs

```diff
@@ -476,18 +478,31 @@ impl<'a> FieldBorrow<'a> {
     }
 }
 
+// Helpful in interacting with external systems during ingestion and querying
+// For example, nanoseconds can overflow.
+#[derive(Clone, Copy, Serialize, Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord)]
+enum TimeUnit {
+    Seconds,
+    Milliseconds,
+    Microseconds,
+    Nanoseconds,
+}
+
 #[derive(Clone, Copy, Serialize, Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub enum FieldType {
     UInt,
+    UInt128,
     Int,
+    Int128,
     Float,
     Boolean,
     String,
     Text,
     Binary,
     Decimal,
-    Timestamp,
-    Date,
+    Timestamp { unit: Option<TimeUnit> },
+    Date { offset: Option<FixedOffset> },
+    Duration { unit: Option<TimeUnit> },
```

Comment:
I don't get it. `Field::Timestamp` and `Field::Duration` are number + unit. How is the unit optional?

Reply:
set the default unit?

Reply:
> set the default unit?

Time unit is always nanoseconds in Dozer now right? Are we going to support `Timestamp` and `Duration` in (for example) microseconds to save the 8 bytes?